# 데이터 load, 확인

In [27]:
import pandas as pd # 데이터 프레임
import numpy as np 
import re # 정규식

## 파일 불러오기

In [30]:
df = pd.read_json('yes24.3.json')

init_df : 아무것도 수정하지 않은 최초의 DataFrame 

In [32]:
df['preview']

0       염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...
1       [목차]\n\n작가의 말\n프롤로그. 3번째 제자의 유서 깊은 가게\n\n1. 주문...
2       지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”...
3       1◆\n이계로 통하는 차원 문\n\n“흐으윽.”\n강성호는 기지개를 켜며 일어났다....
4       * 이 작품은 픽션입니다.\n실제 인물, 단체와는 일절 관계가 없습니다.\n\n\n...
                              ...                        
7769                                                     
7770                                                     
7771                                                     
7772                                                     
7773                                                     
Name: preview, Length: 7774, dtype: object

In [40]:
df.preview[2]

'지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”로 평가받는 영국의 소설가이자 동화작가. 기발한 상상력에 유머와 위트가 더해진 그의 작품은 독자들에게 큰 공감과 위로를 주며 많은 사랑을 받고 있다.\n1975년 영국 사우스요크셔주 셰필드에서 태어났다. 20대 초 절벽 끝에 서서 스스로 생을 마감하려던 순간, 자신의 우울증과 불안장애를 깨달은 작가는 파트너와 가족의 도움을 받아 서서히 건강을 회복했다. 이때부터 소설을 쓰기 시작한 그는 2004년 《영국의 마지막 가족》을 출간하며 소설가로 데뷔했다. 2007년 첫 번째 어린이책 《그림자 숲의 비밀》로 네슬레 어린이도서상과 블루 피터 도서상을 수상했고, 이후 카네기상 최종 후보에 세 번이나 오르며 동화작가로서도 인정을 받았다. 2014년 《휴먼: 어느 외계인의 기록》으로 에드거상 최종 후보에 올랐고, 2015년 우울증을 극복한 과정을 담은 에세이 《살아야 할 이유》가 영국에서 베스트셀러가 되며, “마음 건강에 대하여 우리 시대에 가장 영향력 있는 작가”로 자리매김했다. 그 외에도 20여 편이 넘는 그의 작품은 30여 개 이상의 언어로 번역 출판되었고, 소설 《시간을 멈추는 법》과 동화 《크리스마스로 불리는 소년》은 영화화될 예정이다.\n\n\n옮긴이 노진선\n숙명여대 영어영문학과를 졸업했고, 전문 번역가로 활동하며 다양한 작품들을 번역해왔다. 옮긴 책으로 제닌 커민스의 《아메리칸 더트》, 할런 코벤의 《사라진 밤》, 조디 피코의 《작지만 위대한 일들》, 피터 스완슨의 《죽여 마땅한 사람들》, 요 네스뵈의 《스노우맨》 《레오파드》 《네메시스》 등 ‘형사 해리 홀레 시리즈’와 엘리자베스 길버트의 《먹고 기도하고 사랑하라》 등이 있다.\n\n\n\n\nTHE MIDNIGHT LIBRARY\n\nCopyright © Matt Haig, 2020\nAll rights reserved.\n\nKorean translation copyright © INFLUENTIAL, Inc., 2021\

총 7774개 도서

## 데이터 프레임 정보 확인

In [4]:
init_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7774 entries, 0 to 7773
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   title    7774 non-null   object
 1   preview  7774 non-null   object
dtypes: object(2)
memory usage: 182.2+ KB


미리보기 열에 null(=비어있는거) 값이 없다고 나옴

아무것도 없는 즉, 공백이 있는 상태로 인식함 (빈 값이 아니고 공백이라는 값이 들어있는 상태로 봄)

In [5]:
init_df.loc[init_df.preview==""]

title preview
1281                  아몬드        
1282  [세트] 듄 시리즈 (총6권/완결)        
1283   우리가 빛의 속도로 갈 수 없다면        
1284                종의 기원        
1285               소년이 온다        
...                   ...     ...
7769               완전한 행복        
7770            프로젝트 헤일메리        
7771       1984 (한글판) 019        
7772           구의 증명 [할인]        
7773             대도시의 사랑법        

[6268 rows x 2 columns]

In [29]:
init_df.loc[init_df['preview'] != ""]

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7647                 문어의 아홉 번째 다리   
7648                        백은의 잭   
7652                          빅티켓   
7719                        나쁜 토끼   
7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     [목차]\n\n작가의 말\n프롤로그. 3번째 제자의 유서 깊은 가게\n\n1. 주문...  
2     지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”...  
3     1◆\n이계로 통하는 차원 문\n\n“흐으윽.”\n강성호는 기지개를 켜며 일어났다....  
4     * 이 작품은 픽션입니다.\n실제 인물, 단체와는 일절 관계가 없습니다.\n\n\n...  
...                                                 ...  
7647  디르크 로스만지음| 서경홍옮김\n\n\n\n\n\n\n\n\n\n\n\n\nDER ...  
7648  히가시노 게이고\n일본 최고의 베스트셀러 작가. 1958년 오사카 출생. 오사카 부...  
7652  테릴 리 랭크포드를 위하여\n\n우리는 우리의 삶을 우화처럼 기억한다.\n — 작자...  
7719  와카타케 나나미 若竹七海\n\n일본 코지 미스터리의 여왕. 1963년 도쿄에서 태어...  
7727  이승우\n1959년 전남 장흥에서 태어났다. 1981년 《한국문학》 신인상을 받으며...  

[1506 rows x 2 columns]

6268개가 공백이 들어있음

dropna()는 "nan"값이 포함된 행이나 열을 제거할 수 있음  
공백을 nan으로 바꾸고 dropna()를 쓰면 미리보기가 없는 행을 제거할수 있겠다

In [7]:
nan_df = init_df # 복사본

nan_df : 공백을 nan으로 변환 시킨 데이터 프레임

In [8]:
nan_df.preview = nan_df.preview.replace("", np.nan)
nan_df

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7769                       완전한 행복   
7770                    프로젝트 헤일메리   
7771               1984 (한글판) 019   
7772                   구의 증명 [할인]   
7773                     대도시의 사랑법   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     [목차]\n\n작가의 말\n프롤로그. 3번째 제자의 유서 깊은 가게\n\n1. 주문...  
2     지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”...  
3     1◆\n이계로 통하는 차원 문\n\n“흐으윽.”\n강성호는 기지개를 켜며 일어났다....  
4     * 이 작품은 픽션입니다.\n실제 인물, 단체와는 일절 관계가 없습니다.\n\n\n...  
...                                                 ...  
7769                                                NaN  
7770                                                NaN  
7771                                                NaN  
7772                                                NaN  
7773                                                NaN  

[7774 rows x 2 columns]

# 결측치 제거

In [9]:
erase_df = nan_df # 복사본 

erase_df : 결측치가 제거된 데이터 프레임

In [10]:
erase_df.dropna()

erase_df

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7769                       완전한 행복   
7770                    프로젝트 헤일메리   
7771               1984 (한글판) 019   
7772                   구의 증명 [할인]   
7773                     대도시의 사랑법   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     [목차]\n\n작가의 말\n프롤로그. 3번째 제자의 유서 깊은 가게\n\n1. 주문...  
2     지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”...  
3     1◆\n이계로 통하는 차원 문\n\n“흐으윽.”\n강성호는 기지개를 켜며 일어났다....  
4     * 이 작품은 픽션입니다.\n실제 인물, 단체와는 일절 관계가 없습니다.\n\n\n...  
...                                                 ...  
7769                                                NaN  
7770                                                NaN  
7771                                                NaN  
7772                                                NaN  
7773                                                NaN  

[7774 rows x 2 columns]

inplace=True : 원본에 반영을 하겠다는 뜻

In [11]:
erase_df.dropna(inplace = True)

erase_df

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7647                 문어의 아홉 번째 다리   
7648                        백은의 잭   
7652                          빅티켓   
7719                        나쁜 토끼   
7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     [목차]\n\n작가의 말\n프롤로그. 3번째 제자의 유서 깊은 가게\n\n1. 주문...  
2     지은이 매트 헤이그Matt Haig\n“강렬한 존재감과 위대한 재능을 가진 소설가”...  
3     1◆\n이계로 통하는 차원 문\n\n“흐으윽.”\n강성호는 기지개를 켜며 일어났다....  
4     * 이 작품은 픽션입니다.\n실제 인물, 단체와는 일절 관계가 없습니다.\n\n\n...  
...                                                 ...  
7647  디르크 로스만지음| 서경홍옮김\n\n\n\n\n\n\n\n\n\n\n\n\nDER ...  
7648  히가시노 게이고\n일본 최고의 베스트셀러 작가. 1958년 오사카 출생. 오사카 부...  
7652  테릴 리 랭크포드를 위하여\n\n우리는 우리의 삶을 우화처럼 기억한다.\n — 작자...  
7719  와카타케 나나미 若竹七海\n\n일본 코지 미스터리의 여왕. 1963년 도쿄에서 태어...  
7727  이승우\n1959년 전남 장흥에서 태어났다. 1981년 《한국문학》 신인상을 받으며...  

[1506 rows x 2 columns]

원본에 반영이 됨, 1506개만 남음

처음에 7774개 있었는데 아까 6268개 공백이 있다고 했었기 때문에
1506개만 남게 됨

In [12]:
7774 - 6268

1506

# 미리보기 column : 한글만 포함, 여러 공백 제거, 짧은거 제거

In [13]:
preview_df = erase_df # 복사본

preview_df : 미리보기를 불용어 제거한 데이터 프레임

In [14]:
preview_df.preview = preview_df.preview.str.replace(r"[^가-힣 \s]", " ", regex=True) # 가 각 간 부터 힢 힡 힣 빼고는 다 공백
preview_df.preview = preview_df.preview.str.replace(r"\s+", " ", regex=True) # 연속한 공백을 하나의 공백으로
preview_df.preview = preview_df.preview.str.strip() # 양쪽에 공백 있으면 제거됨

preview_df = preview_df[preview_df.preview.str.len() > 1000] # 1000자가 넘는 데이터만 살려줌

preview_df

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7647                 문어의 아홉 번째 다리   
7648                        백은의 잭   
7652                          빅티켓   
7719                        나쁜 토끼   
7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
7647  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
7648  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
7652  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
7719  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
7727  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1383 rows x 2 columns]

1506개에서 1383개 데이터로 줄어들었음

# 제목 column : [대여] 같은거 제거, 연속된 공백 제거, 중복제거

In [15]:
title_preview_df = preview_df # 복사본

title_preview_df : 미리보기에다가 '제목'까지 전처리 된 데이터 프레임

In [16]:
title_preview_df.title = title_preview_df.title.str.replace(r"\[.+\]", "", regex=True) # [ ] 제거

title_preview_df.title = title_preview_df.title.str.replace(r"\s+", " ", regex=True)

title_preview_df.title = title_preview_df.title.str.strip()

title_preview_df

C:\Users\kissd\anaconda3\lib\site-packages\pandas\core\generic.py:5516: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7647                 문어의 아홉 번째 다리   
7648                        백은의 잭   
7652                          빅티켓   
7719                        나쁜 토끼   
7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
7647  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
7648  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
7652  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
7719  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
7727  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1383 rows x 2 columns]

제목이 같은 데이터를 제거했더니 1383개에서 1308개로 줄어들었음  
근데 책 제목이 같은 거는 지우면 안되고 하나만 남겨야 되는데 ㅋㅋ

## 인덱스 다시 맞춰주자

In [18]:
title_preview_df = title_preview_df.reset_index()

title_preview_df

index                        title  \
0         0                      불편한 편의점   
1         1  달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2         2                   미드나잇 라이브러리   
3         3                      기적의 분식집   
4         4                       유리의 살의   
...     ...                          ...   
1303   7647                 문어의 아홉 번째 다리   
1304   7648                        백은의 잭   
1305   7652                          빅티켓   
1306   7719                        나쁜 토끼   
1307   7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
1303  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
1304  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
1305  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
1306  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
1307  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1308 rows x 3 columns]

drop = True를 안해주면 이 꼴 남

In [19]:
title_preview_df = title_preview_df.reset_index(drop=True)

title_preview_df

index                        title  \
0         0                      불편한 편의점   
1         1  달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2         2                   미드나잇 라이브러리   
3         3                      기적의 분식집   
4         4                       유리의 살의   
...     ...                          ...   
1303   7647                 문어의 아홉 번째 다리   
1304   7648                        백은의 잭   
1305   7652                          빅티켓   
1306   7719                        나쁜 토끼   
1307   7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
1303  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
1304  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
1305  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
1306  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
1307  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1308 rows x 3 columns]

소용없음 그래서 다시 복사본 만들자 (백업느낌)

In [20]:
title_preview_df = preview_df # 복사본

title_preview_df.title = title_preview_df.title.str.replace(r"\[.+\]", "", regex=True) # [ ] 제거
title_preview_df.title = title_preview_df.title.str.replace(r"\s+", " ", regex=True)
title_preview_df.title = title_preview_df.title.str.strip()
title_preview_df = title_preview_df.drop_duplicates(['title']) # 중복된 데이터 제거
title_preview_df = title_preview_df.drop_duplicates(["preview"]) # 중복된 데이터 제거
title_preview_df

C:\Users\kissd\anaconda3\lib\site-packages\pandas\core\generic.py:5516: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
7647                 문어의 아홉 번째 다리   
7648                        백은의 잭   
7652                          빅티켓   
7719                        나쁜 토끼   
7727                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
7647  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
7648  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
7652  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
7719  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
7727  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1285 rows x 2 columns]

중복된 내용을 제거하고 나니 1308개에서 1285개로 줄어들음

In [21]:
title_preview_df = title_preview_df.reset_index(drop=True)

title_preview_df

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
1280                 문어의 아홉 번째 다리   
1281                        백은의 잭   
1282                          빅티켓   
1283                        나쁜 토끼   
1284                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
1280  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
1281  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
1282  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
1283  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
1284  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1285 rows x 2 columns]

In [22]:
final_df = title_preview_df 

final_df : 전처리 마친 데이터 프레임

## 전처리 마친 데이터 json 파일로 저장

In [23]:
pd.DataFrame(final_df).to_json('preprocessed_data.json') #json 파일로 만들기 

## 저장된 파일 확인

In [24]:
pd.read_json('preprocessed_data.json')

title  \
0                         불편한 편의점   
1     달러구트 꿈 백화점 : 잠들어야만 입장 가능합니다   
2                      미드나잇 라이브러리   
3                         기적의 분식집   
4                          유리의 살의   
...                           ...   
1280                 문어의 아홉 번째 다리   
1281                        백은의 잭   
1282                          빅티켓   
1283                        나쁜 토끼   
1284                     욕조가 놓인 방   

                                                preview  
0     염영숙 여사가 가방 안에 파우치가 없다는 걸 알았을 때 기차는 평택 부근을 지나고 ...  
1     목차 작가의 말 프롤로그 번째 제자의 유서 깊은 가게 주문하신 꿈은 매진입니다 한밤...  
2     지은이 매트 헤이그 강렬한 존재감과 위대한 재능을 가진 소설가 로 평가받는 영국의 ...  
3     이계로 통하는 차원 문 흐으윽 강성호는 기지개를 켜며 일어났다 오늘도 지겨운 하루가...  
4     이 작품은 픽션입니다 실제 인물 단체와는 일절 관계가 없습니다 이 책은 주 한국저작...  
...                                                 ...  
1280  디르크 로스만지음 서경홍옮김 이 책의 한국어판 저작권은 모모 에이전시를 통해 사와의...  
1281  히가시노 게이고 일본 최고의 베스트셀러 작가 년 오사카 출생 오사카 부립대학 졸업 ...  
1282  테릴 리 랭크포드를 위하여 우리는 우리의 삶을 우화처럼 기억한다 작자 불명 본 도서...  
1283  와카타케 나나미 일본 코지 미스터리의 여왕 년 도쿄에서 태어나 릿쿄 대학 문학부를 ...  
1284  이승우 년 전남 장흥에서 태어났다 년 한국문학 신인상을 받으며 작품 활동을 시작했다...  

[1285 rows x 2 columns]